crates/bevy_reflect/src/type_registry.rs

```diff
@@ -56,53 +63,94 @@ impl TypeRegistry {
+    ///
+    /// [`TypeId`]: std::any::TypeId
     pub fn get(&self, type_id: TypeId) -> Option<&TypeRegistration> {
         self.registrations.get(&type_id)
     }
 
+    /// Returns a mutable reference to the [`TypeRegistration`] of the type with
+    /// the given [`TypeId`].
+    ///
+    /// If the specified type has not been registered, returns `None`.
+    ///
+    /// [`TypeId`]: std::any::TypeId
     pub fn get_mut(&mut self, type_id: TypeId) -> Option<&mut TypeRegistration> {
         self.registrations.get_mut(&type_id)
     }
 
+    /// Returns a reference to the [`TypeRegistration`] of the type with the
+    /// given name.
+    ///
+    /// If no type with the given name has been registered, returns `None`.
     pub fn get_with_name(&self, type_name: &str) -> Option<&TypeRegistration> {
         self.full_name_to_id
             .get(type_name)
             .and_then(|id| self.get(*id))
     }
 
+    /// Returns a mutable reference to the [`TypeRegistration`] of the type with
+    /// the given name.
+    ///
+    /// If no type with the given name has been registered, returns `None`.
     pub fn get_with_name_mut(&mut self, type_name: &str) -> Option<&mut TypeRegistration> {
         self.full_name_to_id
             .get(type_name)
             .cloned()
             .and_then(move |id| self.get_mut(id))
     }
 
+    /// Returns a mutable reference to the [`TypeRegistration`] of the type with
+    /// the given short name.
+    ///
+    /// If the short name is ambiguous, or if no type with the given short name
+    /// has been registered, returns `None`.
     pub fn get_with_short_name(&self, short_type_name: &str) -> Option<&TypeRegistration> {
         self.short_name_to_id
             .get(short_type_name)
             .and_then(|id| self.registrations.get(id))
     }
 
+    /// Returns the [`TypeData`] of type `T` associated with the given `TypeId`.
+    ///
+    /// If the specified type has not been registered, or if `T` is not present
+    /// in its type registration, returns `None`.
     pub fn get_type_data<T: TypeData>(&self, type_id: TypeId) -> Option<&T> {
         self.get(type_id)
             .and_then(|registration| registration.data::<T>())
     }
 
+    /// Returns an iterator overed the [`TypeRegistration`]s of the registered
+    /// types.
     pub fn iter(&self) -> impl Iterator<Item = &TypeRegistration> {
         self.registrations.values()
     }
 }
 
 impl TypeRegistryArc {
+    /// Takes a read lock on the underlying [`TypeRegistry`].
     pub fn read(&self) -> RwLockReadGuard<'_, TypeRegistry> {
         self.internal.read()
     }
 
+    /// Takes a write lock on the underlying [`TypeRegistry`].
     pub fn write(&self) -> RwLockWriteGuard<'_, TypeRegistry> {
         self.internal.write()
     }
 }
 
+/// A record of data about a type.
+///
+/// This contains the [`TypeId`], [name], and [short name] of the type, as well
+/// as the same data for any type parameters of the type.
```

Comment:
>as well as the same data for any type parameters of the type.

This seems incorrect.
The `data` field contains the information about the reflected traits. For instance, if a type has `#[reflect(Component)]`, the data field will contain an entry at the key `type_id::<ReflectComponent>()` with a boxed`ReflectComponent as dyn TypeData`.